packages/web-frontend/cypress/scripts/generateConfig/configSchema.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { yup, yupUtils } from '@tupaia/utils';
+
+import { SNAPSHOT_TYPES } from '../../constants';
+
+const { oneOfType, oneOrArrayOf } = yupUtils;
+
+const shape = shapeConfig => yup.object(shapeConfig).noUnknown();
+
+const arrayOf = values => yup.array().of(yup.mixed().oneOf(values));
+
+const strings = yup.array().of(yup.string());
+const stringOrStrings = oneOrArrayOf(yup.string());
+const stringsOrObjects = yup.array().of(oneOfType([yup.string(), yup.object()]));
+
+export const configSchema = shape({
```

Comment:
@rohan-bes we have discussed in the past about config validation... You may want to have a look at the validation approach taken here (https://github.com/beyondessential/tupaia/pull/2694/commits/06a34f8403ec20c78f40d1176ef67ad815048a08)

I used [yup](https://www.npmjs.com/package/yup), which is a cute validation library also used in Tamanu. Other options considered (briefly):
* [joi](https://github.com/sideway/joi), which seems like a more powerful version of `yup`
* [ajv](https://www.npmjs.com/package/ajv), which is a [JSON schema](https://json-schema.org/) implementation. I really like how VSCode uses JSON schema to enable config validation and autocompletion (eg when editing `eslint` or `jest` config).

For our purposes here and given that we already use `yup` in Tamanu, I decided to go with yup (had a brief discussion with @edmofro about this too).

You may be interested in the above for introducing config validation in `report-server`, and I sense we may want to leverage this in the Admin Panel Viz Builder. Apart from validation, `yup` also handles casting (filling in empty fields, converting between types etc). Let me know if you need more info!